zaza/openstack/charm_tests/kerberos/setup.py

```diff
@@ -0,0 +1,239 @@
+
+    dump_file = "krb5.keytab"
+    remote_file = "/etc/krb5.keytab"
+    host_keytab_path = '/home/ubuntu/krb5.keytab'
+    with tempfile.TemporaryDirectory() as tmpdirname:
+        tmp_file = "{}/{}".format(tmpdirname, dump_file)
+        logging.info("Retrieving {} from {}.".format(remote_file,
+                                                     kerberos_server.name))
+        zaza.model.scp_from_unit(
+            kerberos_server.name,
+            remote_file,
+            tmp_file)
+
+        logging.info("SCP {} to {} on {}.".format(tmp_file,
+                                                  host_keytab_path,
+                                                  ubuntu_test_host.name))
+        zaza.model.scp_to_unit(
+            ubuntu_test_host.name,
+            tmp_file,
+            host_keytab_path)
+
+    dump_file = "krb5.conf"
+    remote_file = "/etc/krb5.conf"
+    temp_krb5_path = "/home/ubuntu/krb5.conf"
+    with tempfile.TemporaryDirectory() as tmpdirname:
+        tmp_file = "{}/{}".format(tmpdirname, dump_file)
+        logging.info("Retrieving {} from {}".format(remote_file,
+                                                    kerberos_server.name))
+        zaza.model.scp_from_unit(
+            kerberos_server.name,
+            remote_file,
+            tmp_file)
+
+        logging.info("SCP {} to {} on {}.".format(tmp_file,
+                                                  temp_krb5_path,
+                                                  ubuntu_test_host))
+        zaza.model.scp_to_unit(
+            ubuntu_test_host.name,
+            tmp_file,
+            temp_krb5_path)
+        logging.info('Moving {} to {} on {}.'.format(temp_krb5_path,
+                     remote_file, ubuntu_test_host.name))
+        zaza.model.run_on_unit(ubuntu_test_host.name, ('sudo mv {} {}'.
+                               format(temp_krb5_path, remote_file)))
+
+
+def install_apt_packages_on_ubuntu_test_host():
+    """Install apt packages on a zaza unit."""
+    ubuntu_test_host = zaza.model.get_units('ubuntu-test-host')[0]
+    packages = ['krb5-user', 'python-openstackclient', 'python-pip', 'gcc',
+                'python-dev', 'libkrb5-dev']
+    for package in packages:
+        logging.info('Installing {}'.format(package))
+        result = zaza.model.run_on_unit(ubuntu_test_host.name,
+                                        "apt install {} -y".format(package))
+        assert result['Code'] == '0', result['Stderr']
+
+
+def install_python_packages_on_ubuntu_test_host():
+    """Install python packages on a zaza unit."""
+    ubuntu_test_host = zaza.model.get_units('ubuntu-test-host')[0]
+    package = 'keystoneauth1[kerberos]'
+    logging.info('Installing {}'.format(package))
+    result = zaza.model.run_on_unit(ubuntu_test_host.name,
+                                    "pip install {} ".format(package))
+    assert result['Code'] == '0', result['Stderr']
+
+
+def run_all_tests():
```

Comment:
I think the run_all_tests method name is a bit misleading, as they are all really configuration steps, and it doesn't actually include the most meaningful test here (CharmKeystoneKerberosTest).

You could either rename this method, or include ALL tests. I would lean toward a rename, mainly because of the Zaza test.yaml primitives of .configure and .test.

Reply:
True, rename it is!